openfe/setup/methods/rbfe_methods.py

```diff
@@ -19,95 +19,195 @@
-    nonbonded_electrostatics
-      which nonbonded electrostatics method to use, currently only 'PME'
-      allowed
-    nonbonded_cutoff
-      range of nonbonded forces
-    constraints
-      which bonds to apply constraints to
-    rigid_water
-      whether apply rigid constraints to water molecules, default True
+    nonbonded_method : str
+        Which nonbonded electrostatic method to use, currently only PME
+        is supported.
+    nonbonded_cutoff : float * unit.nanometer
+        Cutoff value for short range interactions.
+        Default 1.0 * unit.nanometer.
+    constraints : str
+        Which bonds and angles should be constrained. Default None.
+    rigid_water : bool
+        Whether to apply rigid constraints to water molecules. Default True.
+    hydrogen_mass : float
+        How much mass to repartition to hydrogen. Default None, no
+        repartitioning will occur.
     """
     class Config:
         arbitrary_types_allowed = True
 
-    # mapping of component name to forcefield path(s)
-    forcefield: Dict[str, Union[List[str, ...], str]]
-    nonbonded_electostatics = 'PME'
-    nonbonded_cutoff = 0.9 * unit.nanometer
-    constraints = 'HBonds'
+    nonbonded_method = 'PME'
+    nonbonded_cutoff = 1.0 * unit.nanometer
+    constraints = Union[str, None] # Usually use HBonds
     rigid_water = True
+    remove_com = True # Probably want False here
+    hydrogen_mass = Union[float, None]
 
 
-class StatePoint(BaseModel):
-    """Description of the temperature and pressure to model"""
-    class Config:
-        arbitrary_types_allowed = True
+class SolventBoxSettings(BaseModel):
+    """Settings for the creation of the simulation box, inc. solvation
+    and addition of ions
 
-    temperature = 298.15 * unit.kelvin
-    pressure = 1 * unit.bar
+    Attributes
+    ----------
+    model : str
+      The water model to use. Default 'tip3p'.
+    padding : float
+      Padding distance from the solute to the edge of the box. Default None.
+    ionic_strength : float
+      Total concentration of ions (both positive and negative) to add,
+      excluding the ions required to neutralize the system.
+      Default 0 * unit.molar.
+    num_added : int
+      Total number of molecules (waters and ions) to add. Default None.
+    positive_ion : str
+      Type of positive ion to add, only monovalent ions are supported.
+      Default 'Na+'.
+    negative_ion : str
+      Type of negative ion to add, only monovalent ions are supported.
+      Default 'Cl-'.
+    neutralize : float
+      Whether to add ions to neutralize the system. Default True.
+    box_vectors : tuple of Vec3
+      User defined box vectors to fill with water. Default None.
+    box_size : Vec3
+      User defined box size to fill with water. Default None.
+    """
+    model = 'tip3p'
+    padding = Union[float, None]
+    ionic_strength = 0 * unit.molar
+    num_added = Union[int, None]
+    positive_ion = 'Na+'
+    negative_ion = 'Cl-'
```

Comment:
Strength and identity of the ions are covered by `SolventComponent`

Reply:
I think we still need something more specific to this solvation method's parameters. I was working on this:

```python
class OpenMMSolvation:
    model: str = 'tip3p'
    boxSize: Tuple[float, float, float]
    padding: float = 1.2
    positiveIon: str = "Na+"
    negativeIon: str= "Cl-"
    ionicStrength: float = 0.0
    
    @classmethod
    def from_chemical_system(cls, chemical_system, model, solvent_component='solvent'):
        # should model default to tip3p here?
        gufe_solvent = chemical_system[solvent_component]
        
        
        pass
    
    def add_solvent(self, unsolvated_topology, unsolvated_positions):
        modeller = openmm.app.Modeller(unsolvated_topology, unsolvated_positions)
        modeller.addSolvent(
            model=self.model,
            boxsize=self.boxsize,
            padding=self.padding,
            positiveIon=self.positiveIon,
            negativeIon=self.negativeIon,
            ionicStrength=self.ionicStrength
        )


```

Reply:
Basically, we need an adapter between gufe solvent and the OpenMM `Modeller.addSolvent` method

Reply:
Ok that clarifies things - everything here but the model should be defined in the SolventComponent (even if it currently isn't).

In that case let's move model to force field (technically it's already defined but for now let's keep it different). and get rid of this class completely?

Reply:
Sorry didn't see @dwhswenson's message. Need to rethink that one, but can we say that whatever we need to bridge things needs to exist "somewhere else"?

Reply:
The identity & strength of ions is covered in `Component`.  If you want to tweak that to tighten up the checks, go over to `gufe`

Reply:
I think this is the place for it. "The algorithm we use to solvate the system" is component that should be modular. The `SolventComponent` defines some characteristics of the final product, but the algorithm itself could change.

Again, what we're doing here is nounifying the verbs.

Reply:
@richardjgowers So instead of my classmethod `from_chemical_system`, you want that to be `__init__` and make the kwargs for `addSolvent` be properties?

Reply:
When do we wrap in defaults? This would require that gufe objects have defaults, wouldn't it?

Reply:
I've moved this out for now, I'll let this discussion run alongside me doing the other settings. Can always bring things back in later.